---
title: "CLEANING_TRANSFO_HALASEN"
output: html_document
date: "2023-04-12"
editor_options: 
  chunk_output_type: console
---

*NOTES* TOASK : ctrl + F this and search - new variable : upper - lower - resolve otu pb - *LIBRARY*

```{r packages}
library(devtools) # for install_github etc. 
library(ggplot2) # Create Elegant Data Visualisations Using the Grammar of Graphics
library(dplyr) # A Grammar of Data Manipulation
library(tidyverse) # A Grammar of Data Manipulation
library(vegan) # ecological statistical analysis
library(phyloseq) # Handling and analysis of high-throughput microbiome census data 
library(forcats) # Tools for Working with Categorical Variables
library(MiscMetabar) # Analyze and visualize metabarcoding data
library(circlize) # circular visualization
library(ggVennDiagram) # venn diagrams 
library(phyloseq.extended) # phyloseq packages with more options
library (microbiomeutilities) # for rarefaction curves
library(multcompView) # Visualizations of Paired Comparisons
library(plyr) # tools for cleaning 
library(pals) # color palette
library(ade4) # Analysis Data Ecological Environmental in the framework of Euclidean Exploratory methods
library(lme4) # Linear, generalized linear, and nonlinear mixed models
library(lmtest) # Diagnostic Checking in Regression Relationships
library(reshape2)
library(data.table)
library(viridis) # palettes and themes
library(hrbrthemes) # palettes and themes
library(emmeans) # tukey test for GLM
library(ISLR) # McFaddenâ€™s R-Squared for GLM
library(ape) # Analyses of Phylogenetics and Evolution
library(sjPlot) # for glmm visualization
# undersampling lib
library(purrr) # purrr: Functional Programming Tools
library(randomForest) # randomForest implements Breiman's random forest algorithm (based on Breiman and Cutler's original Fortran code) for classification and regression
library(caret)
library(phylosmith) # A supplementary package to build on the phyloseq package
library(brms) # Bayesian Regression Models using 'Stan'
library(glmmTMB) # Fit a generalized linear mixed model (GLMM) using Template Model Builder (TMB).
library(gtsummary)
library(pairwiseAdonis) # pari-wise analysis for PermANOVAs 
library(gplots)
library(FUNGuildR)
```
```{r color palette}
pals::pal.bands(alphabet, alphabet2, cols25, glasbey, kelly, polychrome, 
  stepped, tol, watlington,
  show.names=FALSE)

funky <- grDevices::colorRampPalette(c("#A6CEE3","#1F78B4","#B2DF8A",
"#33A02C","#FB9A99","#E31A1C",
"#FDBF6F","#FF7F00","#CAB2D6",
"#6A3D9A","#FFFF99","#B15928"))
```
```{r functions}
outersect <- function(x, y) {
  sort(c(setdiff(x, y),
         setdiff(y, x)))
} # extract uncommon values between x - y 

stde <- function (x) {sqrt(var(x))/length(x)}

give.n <- function(x){
   return(c(y = mean(x), label = length(x)))
}
```
```{r dummy calc}
# hellinger : sum of reads per sample = 1 ? (as same as relative abundance ?)

#abund1 <- c(sample(0:250, n = 10))

# function if()
x <- 0
if (x < 0) {
  print("Negative number")
} else if (x > 0) {
  print("Positive number")
} else {
  print("Zero")
}

round_df <- function(x, digits) {
    # round all numeric variables
    # x: data frame 
    # digits: number of digits to round
    numeric_columns <- sapply(x, mode) == 'numeric'
    x[numeric_columns] <-  round(x[numeric_columns], digits)
    x
}

# if() is used for a single value
# to check for a vector: use ifelse instead
# can use ifelse(test, yes, ifelse(test, yes, no)) nested 
```
```{r objects}
coltreatment <- c("FE" = "khaki2",
                  "SC" = "coral2",
                  "FC" = "springgreen4") # palette for treatment

coltreatment_raref <- c("FE" = "orange",
                                "SC" = "red3",
                                "FC" = "springgreen4")

order_level <- c("FC", "FE", "SC") # ordination of factors (aes = factor(x, levels = order_level))
order_level_mg <- c("2012FC", "2012FE", "2012SC", "2018FC", "2018FE", "2018SC")
order_level_simpl <- c("FC_12", "FE_12", "SC_12", "FC_18", "FE_18", "SC_18")

plot_list <- as.factor(ecm_phy@sam_data$plot) %>% droplevels() %>% levels() %>% str_sort(numeric = T) %>% print() # list of plots
```

*DATA IMPORTATION AND MODIF*

```{r raw_data}
setwd("H:/REDACTION/RESULTS/cutting_ecm/TRANSF_DATA2") 
# SLU: H:/REDACTION/RESULTS/cutting_ecm/TRANSF_DATA2
# HOME: C:/Users/Utilisateur/OneDrive/STAGE-2023_SUEDE/cutting_ecm-main_branch/TRANSF_DATA2
#data run

# fruitbody dataset
fruit <- read.table("fruit.txt", header = T, sep="\t",na.strings=T,dec=".") 
rownames(fruit) <- fruit$X # putting X column into rownames
fruit <- fruit[, -1] # supressing obsolete column
colnames(fruit) <- gsub("X", "", colnames(fruit)) # supressing "X" from colnames 

# sequencing run metadata dataset
run_data <- read.table("run_data.txt", header=T, sep="\t",na.strings=T,dec=".")
colnames(run_data)

# all clusters
raw_clust <- read.table("raw_clusters.txt",header=T,sep="\t",na.strings=T,dec=".") %>%
  arrange(desc(Cluster_Size)) # importing and arrange by decreasing cluster size
raw_clust[raw_clust == ""] <- NA # putting NA when empty
raw_clust[, 4] <- as.character(raw_clust[, 4]) # reference as.character
raw_clust$Cluster_ID <- gsub('scata5670_', 'c', raw_clust$Cluster_ID) # replace character occurrence for every cells
rownames(raw_clust) <- raw_clust$Cluster_ID # attributing each row to their cluster_id (OTU name) 

order <- raw_clust$Cluster_ID 
order <- gsub('c','', raw_clust$Cluster_ID) # creating a numerical vector order (by deleting the "c"), so we can arrange later by number
order <- as.numeric(order)
raw_clust <- data.frame(cbind(order, raw_clust)) # binding string and df
raw_clust[, "order"] <- as.numeric(raw_clust[, "order"]) # transf. in num. variable

# subsetting fungi X non-fungi
raw_f_clust <- subset(raw_clust, !grepl("PLANT|NF", raw_clust[, 5]))
raw_nf_clust <- subset(raw_clust, grepl("PLANT|NF", raw_clust[, 5]))

# taxonomic and ecological traits table
# only the 1st 1000 clusters 
taxeco_id <- read.table("tax_table.txt",header=T,sep="\t") # change #N/A in excel in NA
length(taxeco_id) # number of columns 
length(rownames(taxeco_id)) # number of rows (ie clusters, 1st 1000 fungi)
taxeco_id[, c(3:length(taxeco_id))] <- lapply(taxeco_id[, c(3:length(taxeco_id))], as.factor) # transform character strings into factors
rownames(taxeco_id) <- taxeco_id$Cluster_ID # rownames of corresponding clusters
```

```{r test funguild}
attrib_raw <- funguild_assign(raw_clust, tax_col = "Genus")
# could be interesting 
```

```{r tax-eco table manipulation }
# creating the variable "ID_TYPE" (auto or manual) for the OTU identification. 
## for this, we check the match between the OTU_ID raw_f_clust (ie the whole dataset) and taxeco (ie manually reviewed table)
id_tax <- raw_f_clust %>%
  filter(Cluster_ID %in% taxeco_id$Cluster_ID) %>%
  arrange(order) # filtering taxtable identified manually
id_tax <- cbind("MANUAL_ID", id_tax)

unid_tax <- raw_f_clust %>%
  filter(!Cluster_ID %in% taxeco_id$Cluster_ID) %>%
  arrange(order) # filtering taxtable not identified 

tax_f <- dplyr::bind_rows(id_tax, unid_tax) # merging "id_tax" and "unid_tax" together
colnames(tax_f)[1] <- "ID_TYPE"
tax_f$ID_TYPE[is.na(tax_f$ID_TYPE)] <- "AUTO_ID" # replacing the "ID_TYPE" column with new value

taxeco_id <- cbind("MANUAL_ID", taxeco_id) 
colnames(taxeco_id)[1] <- "ID_TYPE"

# id_tax and taxeco_id should be identical 
match_id <- taxeco_id$Cluster_ID %in% id_tax$Cluster_ID
as.factor(match_id) %>% 
  levels()
which(match_id == "FALSE", arr.ind = T) # mismatch at row 240
## not same length row between id_tax (998) and taxeco (999): not normal, can cause offset
## can be explained by forgetting an OTU during manual process that lead to offset everything.

# cluster is identified as "Fungi": no relevance in our analysis, we can then delete it
taxeco_id <- taxeco_id[-240,] # deletion of row 240

match_id <- taxeco_id$Cluster_ID %in% id_tax$Cluster_ID # checking if resolved
as.factor(match_id) %>% 
  levels()
which(match_id == "FALSE", arr.ind = T) # good. no more mismatch

## We need to merge the "tax_f" which includes whole fungal community but is lacking tax-ecological traits (only attributed from 1st 1000 OTUs), with "taxeco_id" that includes their reference from SCATA with the dataset manually IDed + FunGuild attribution. 
taxeco_id$Cluster_Size <- id_tax$Cluster_Size # reattributing the proper cluster_size values

taxeco2 <- merge(tax_f, taxeco_id, all = T) %>%
  arrange(order) # merging taxeco and idtaxeco  

rownames(taxeco2) <- taxeco2$Cluster_ID

# attribution of automatic ID (ie reference) for OTU that were not manually reviewed
which(taxeco2[,"ID_TYPE"] == "MANUAL_ID") %>% length() # finding the rows NOT concerned
taxeco2$OTU_ID <- as.character(taxeco2$OTU_ID) # putting as character for next line
taxeco2$OTU_ID <- c(taxeco2$OTU_ID[1:998],  taxeco2$Reference[999:length(taxeco2$Reference)]) # merging reference and otu_id together
# AUTO_ID OTUs have tax-ecological traits column from FunGuild with NAs values but have now value in OTU_ID column
  
# subsetting interesting variables by keeping only tax-ecological traits 
colnames(taxeco2)
taxeco <- taxeco2[, c(17:24,1)] 
taxeco <- sapply(taxeco, as.character) 
rownames(taxeco) <- rownames(taxeco2) # cluster name in row
colnames(taxeco)
colnames(taxeco)[8] <- "Fruitbody" # change variable name
```
```{r sample data}
# sample data
sam <- read.table("sample_table.txt",header=T,sep="\t",na.strings = T) 
sam_no_match <- read.table("samples_no_match.txt",header=T,sep="\t") 

sam[sam==""] <- NA # filling the blanck by NAs
rownames(sam) <- sam$sample_id # after transposing, resolving pb with headers

# subsetting design samples per site 
sam_hala <- subset(sam, sam$site == "Halasen") # 61 samples # 64 initially!
# missing 2 SC + 1 FC
sam_hala5p <- subset(sam, sam$site == "Halasen_p_5")
sam_fanga <- subset(sam, sam$site == "Fangamon")
sam_controls_na <- filter(sam, !site %in% c("Halasen", "Fangamon", "Halasen_p_5")) # 19 samples (or controls)
sam_all <- filter(sam, site %in% c("Halasen", "Fangamon", "Halasen_p_5"))
sam <- sam_all
```
```{r changing levels of factor for sam}
# for Halasen SC 
factor(sam$specification) %>% levels()
sam$treatment_code <- plyr::revalue(sam$specification,
                           c("edge_north"="edge", 
                             "edge_south"="edge",
                             "kant_norra"="edge", 
                             "middle_lower"="middle", 
                             "middle_upper"="middle",
                             "midle"="middle"))
as.factor(sam$treatment_code) %>% droplevels() %>% levels()

sam$specification_value <- plyr::revalue(sam$specification,
                           c("edge_north"="N", 
                             "edge_south"="S",
                             "kant_norra"="N", 
                             "middle_lower"="LO", 
                             "middle_upper"="UP",
                             "midle"= NA,
                             "middle"= NA))

sam <- within(sam, treatment_code[treatment_code=="middle" & treatment=="forest"] <- "FC") 

sam$treatment_code <- plyr::revalue(sam$treatment_code,
                           c("middle"="SC", 
                             "edge"="FE"))
as.factor(sam$treatment_code) %>% droplevels() %>% levels()


# for Halasen thinning + Fangamon

sam <- within(sam, treatment_code[site == "Halasen_p_5" & treatment=="control"] <- "FC") 
sam <- within(sam, treatment_code[site == "Halasen_p_5" & treatment=="clearcut"] <- "CC") 
sam <- within(sam, treatment_code[site == "Halasen_p_5" & treatment=="p_5"] <- "TH")
sam <- within(sam, treatment_code[site == "Fangamon" & treatment=="forest_control"] <- "FC") 
sam <- within(sam, treatment_code[site == "Fangamon" & treatment=="clearcut"] <- "CC") 
sam <- within(sam, treatment_code[site == "Fangamon" & treatment=="shelter_without_burn"] <- "TH_NB") 
sam <- within(sam, treatment_code[site == "Fangamon" & treatment=="shelter_burn"] <- "TH_B") 
###

## tables
table(sam$site, sam$treatment_code, sam$year)

```
```{r random resampling -> balanced design}
sam12 <- subset(sam, year == "2012")
sam18 <- subset(sam, year == "2018")
table(sam12$treatment_code)
table(sam18$treatment_code)
table(sam$treatment_code)
sam$treatment_code <- as.factor(sam$treatment_code)

# Set the seed for reproducibility
set.seed(123)

## FC 2012
# Subsample 8 observations from the "FC" category
for_subset12 <- sam12[sam12$treatment_code == "FC", ]
for_subset12 <- for_subset12[sample(nrow(for_subset12), 8), ]

# Combine the subsampled "FC" category with the other categories
equal_sample12 <- rbind(
  sam12[sam12$treatment_code == "SC", ],
  sam12[sam12$treatment_code == "FE", ],
  for_subset12
)

# Generate the table of sample sizes for each category
table(equal_sample12$treatment_code)
sam12 <- equal_sample12

## FC 2018
# Subsample 8 observations from the "FC" category
for_subset18 <- sam18[sam18$treatment_code == "FC", ]
for_subset18 <- for_subset18[sample(nrow(for_subset18), 8), ]

# Combine the subsampled "FC" category with the other categories
equal_sample18 <- rbind(
  sam18[sam18$treatment_code == "SC", ],
  sam18[sam18$treatment_code == "FE", ],
  for_subset18
)

# Generate the table of sample sizes for each category
table(equal_sample18$treatment_code)
sam18 <- equal_sample18

sam_1218 <- rbind(sam12, sam18) # do the subsampling after physeq creation 
```

```{r otu table}
otu_all <- read.table("otu_table.txt",header=F,sep="\t",na.strings = T) 
otu_all = setNames(data.frame(t(otu_all[,-1])), otu_all[,1]) # transpose + 1st column (ie samples set as header
## samples = columns, otu abundance = rows
colnames(otu_all)[1] <- "Cluster_ID" # after transposing, resolving pb with headers
nrow(otu_all) # total number of all clusters (ie fungi ecm or no-ecm, plants)
rownames(otu_all) <- otu_all$Cluster_ID
otu_all <- otu_all[-1] # deleting 1st column (cluster_id is now in rownames)
colnames(otu_all)
cluster_names_otu_all <- rownames(otu_all)

# removing clusters with 0 rowsums 
## relics of PCR mismatch tag (still amplified)
## but clusters still there after the exclusion of mismatch tags (eg ITS7-49 x ITS4-48)
otu_all <- sapply(otu_all, as.numeric) %>% as.data.frame()
rownames(otu_all) <- cluster_names_otu_all
which(rowSums(otu_all) == 0) %>% length() # clust with 0 reads = 557 (4050 - 557 = 3493)
otu <- otu_all %>%
  filter(!rowSums(otu_all) ==0) # suppression of these 557 clusters from otu matrix
nrow(otu)
ncol(otu)

# controls
not_samples_kept <- c("sITS-standard2", "sITS-standard1", "sPCR_neg_1", "sPCR_neg_2", "sQ2") # cf MM_RESULTS.Rmd ```{r DESIGN SAMPLES}  to understand how they are found 

samples_kept_pq <- otu[, colnames(otu) %in% not_samples_kept] # table of control samples and clusters in row. we want to identity and deleted clusters with reads in these. (potential contaminations)

clusters_in_controls <- which(samples_kept_pq > 0, arr.ind = T) %>% 
rownames() %>%  as.factor() %>% levels() %>% print() # 12 clusters found in controls samples
otu_samples_out <- otu[!(rownames(otu) %in% not_samples_kept),] # suppressing these samples from otu matrix
ncol(otu_samples_out)
nrow(otu_samples_out) 

# excluding NF clusters to rarefy only with fungi
otu_fungi_all <- otu_samples_out %>%
  filter(rownames(otu_samples_out) %in% rownames(taxeco)) # taxeco has only fungi clusters
ncol(otu_fungi_all)
nrow(otu_fungi_all)

# creating otu_fungi_hala only for hala to rarefy on the unique same site

otu_fungi_t0 <- t(otu_fungi_all) %>% as.data.frame()
# halasen sc
otu_fungi_hala <- otu_fungi_t0 %>%
  filter(rownames(otu_fungi_t0) %in% rownames(sam_hala)) 

# halasen 5p
otu_fungi_hala5p <- otu_fungi_t0 %>%
  filter(rownames(otu_fungi_t0) %in% rownames(sam_hala5p)) 

# fangamon
otu_fungi_fanga <- otu_fungi_t0 %>%
  filter(rownames(otu_fungi_t0) %in% rownames(sam_fanga)) 

###
otu_fungi <- t(otu_fungi_hala)
# transposing back
ncol(otu_fungi)
nrow(otu_fungi) # 3493 - 3411 = 82 clusters suppressed

sam_hala[which(rownames(sam_hala) %in% rownames(otu_fungi_t0) == FALSE), ]
##

clusters_in_controls <- which(samples_kept_pq >0, arr.ind = T) %>% 
  rownames() %>% as.factor() %>% levels() %>% print() 


# transfo into matrix
otu1 <- matrix(as.numeric(unlist(otu_fungi)),
              ncol = ncol(otu_fungi),
              nrow = nrow(otu_fungi)) # puting array numeric and transforming in matrix for further functions 

# re-attribution (after being deleted from matrix transfo)
rownames(otu1) <- rownames(otu_fungi)
colnames(otu1) <- colnames(otu_fungi)
```

```{r otu <1% suppresion all}
# transpo
otu_fungi_hala <- t(otu_fungi_hala)
otu_fungi_hala5p <- t(otu_fungi_hala5p)
otu_fungi_fanga <- t(otu_fungi_fanga)

# transfo into matrix
otu1 <- matrix(as.numeric(unlist(otu_fungi_hala)),
              ncol = ncol(otu_fungi_hala),
              nrow = nrow(otu_fungi_hala)) 
rownames(otu1) <- rownames(otu_fungi_hala)
colnames(otu1) <- colnames(otu_fungi_hala)

otu2 <- matrix(as.numeric(unlist(otu_fungi_hala5p)),
              ncol = ncol(otu_fungi_hala5p),
              nrow = nrow(otu_fungi_hala5p)) 
rownames(otu2) <- rownames(otu_fungi_hala5p)
colnames(otu2) <- colnames(otu_fungi_hala5p)

otu3 <- matrix(as.numeric(unlist(otu_fungi_fanga)),
              ncol = ncol(otu_fungi_fanga),
              nrow = nrow(otu_fungi_fanga)) 
rownames(otu3) <- rownames(otu_fungi_fanga)
colnames(otu3) <- colnames(otu_fungi_fanga)

# suppression of <1 %
otu_relative1 <- prop.table(otu1, margin = 2) # margin = 2 -> column sum and relative abundances
otu_001_transf1 <- ifelse(otu_relative1 > 0.01,otu1,0) # creating a new matrix where <1% reads are transformed into 0
otu_001_1 <- otu_001_transf1

otu_relative2 <- prop.table(otu2, margin = 2) 
otu_001_transf2 <- ifelse(otu_relative2 > 0.01, otu2, 0) 
otu_001_2 <- otu_001_transf2

otu_relative3 <- prop.table(otu3, margin = 2) 
otu_001_transf3 <- ifelse(otu_relative3 > 0.01, otu3, 0) 
otu_001_3 <- otu_001_transf3
```
```{r otu rarefaction all}
# rarefaction based on all 
otu_t01 <- t(otu_001_1)
otu_t1 <- matrix(as.numeric(unlist(otu_t01)),
              ncol = ncol(otu_t01),
              nrow = nrow(otu_t01))
rownames(otu_t1) <- rownames(otu_t01)
colnames(otu_t1) <- colnames(otu_t01) 

otu_t02 <- t(otu_001_2)
otu_t2 <- matrix(as.numeric(unlist(otu_t02)),
              ncol = ncol(otu_t02),
              nrow = nrow(otu_t02))
rownames(otu_t2) <- rownames(otu_t02)
colnames(otu_t2) <- colnames(otu_t02)

otu_t03 <- t(otu_001_3)
otu_t3 <- matrix(as.numeric(unlist(otu_t03)),
              ncol = ncol(otu_t03),
              nrow = nrow(otu_t03))
rownames(otu_t3) <- rownames(otu_t03)
colnames(otu_t3) <- colnames(otu_t03)

otu_rarefied1 <- as.data.frame(rrarefy(otu_t1, min(rowSums(otu_t1))))
otu_rarefied2 <- as.data.frame(rrarefy(otu_t2, min(rowSums(otu_t2)))) 
otu_rarefied3 <- as.data.frame(rrarefy(otu_t3, min(rowSums(otu_t3)))) 

otu_rarefied1 <- t(otu_rarefied1)
otu_rarefied1 <- matrix(as.numeric(unlist(otu_rarefied1)),
              ncol = ncol(otu_rarefied1),
              nrow = nrow(otu_rarefied1))
rownames(otu_rarefied1) <- rownames(otu_001_1)
colnames(otu_rarefied1) <- colnames(otu_001_1)

otu_rarefied2 <- t(otu_rarefied2)
otu_rarefied2 <- matrix(as.numeric(unlist(otu_rarefied2)),
              ncol = ncol(otu_rarefied2),
              nrow = nrow(otu_rarefied2))
rownames(otu_rarefied2) <- rownames(otu_001_2)
colnames(otu_rarefied2) <- colnames(otu_001_2)

otu_rarefied3 <- t(otu_rarefied3)
otu_rarefied3 <- matrix(as.numeric(unlist(otu_rarefied3)),
              ncol = ncol(otu_rarefied3),
              nrow = nrow(otu_rarefied3))
rownames(otu_rarefied3) <- rownames(otu_001_3)
colnames(otu_rarefied3) <- colnames(otu_001_3)

# merging together
otu_rarefied <- cbind(otu_rarefied1, otu_rarefied2, otu_rarefied3)

# data properties BEFCE - AFTER rarefaction
hist(colSums(otu_t)) # before raref
hist(colSums(otu_rarefied)) # after raref
#
summary(colSums(otu_001))
rd_sps_raref <- data.frame(colnames(otu_001), colSums(otu_001), colSums(otu_rarefied)) # read sum per sample
colnames(rd_sps_raref) <- c("sample_id", "sumreads_notrarefied", "sumreads_otu_rarefied")
summary(colSums(otu_001))
summary(colSums(otu_rarefied))
str(rd_sps_raref)

rd_sps_raref_gg <- reshape2::melt(rd_sps_raref[,2:3])
ggplot(rd_sps_raref_gg, aes(x = value, color = variable, fill = variable))+
  geom_histogram(aes(y = ..count..), alpha = 0.4, position = "identity")+
  geom_density(alpha = 0.3)+
  scale_y_continuous(name = "nb of samples")+
  scale_x_continuous(name = "read sum per sample", breaks = seq(0,3000 ,by = 200))+
  ggtitle("Number of samples per read sum") # raref worked good
```

*CREATION OF PHYSEQ OBJECTS*
```{r physeq objects}
# creating the physeq objects 
otu_phy_rarefied <- otu_table(otu_rarefied, taxa_are_rows = TRUE)
otu_phy_notrarefied <- otu_table(otu_notrarefied, taxa_are_rows = TRUE)
otu_phy_hellinger <- otu_table(otu_hellinger, taxa_are_rows = TRUE)
otu_phy_relative <- otu_table(otu_relative, taxa_are_rows = TRUE)
otu_phy_001kept <- otu_table(otu1, taxa_are_rows = TRUE)
taxeco_phy <- phyloseq::tax_table(taxeco)
sam_phy <- sample_data(sam)

# general physeq object
physeq <- phyloseq(otu_phy_rarefied,taxeco_phy,sam_phy) %>% 
clean_pq(remove_empty_samples = FALSE,remove_empty_taxa = T, clean_samples_names = FALSE)
physeq_rarefied <- physeq
physeq_notrarefied <- phyloseq(otu_phy_notrarefied,taxeco_phy,sam_phy) %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)
# 2822 taxa removed from the <1% relative abundance suppresion. 

physeq_hellinger <- phyloseq(otu_phy_hellinger,taxeco_phy,sam_phy) %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)
physeq_relative <- phyloseq(otu_phy_relative,taxeco_phy,sam_phy) %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)
physeq_001kept <- phyloseq(otu_phy_001kept,taxeco_phy,sam_phy) 
# 10 taxa = erasing taxa from the control samples (among 13, 3 of them are probably also in others informative samples)
clusters_in_controls # controls to understand

# verif sample N
sample_names(physeq)
length(sample_names(physeq)) # 154 samples, as expected 
```

```{r taxa attribution mistakes}
# reattribution of correct names
physeq@tax_table[physeq@tax_table == "Gautieria_"] <- "Gautieria"
physeq@tax_table[physeq@tax_table == "Piloderma__"] <- "Piloderma"
physeq@tax_table[physeq@tax_table == "Sebacina__"] <- "Sebacina"
physeq@tax_table[physeq@tax_table == "Inocybe_geophylla"] <- "Inocybe_geophylla_(coll.)"

# changing variable levels
levels(as.factor(physeq@tax_table[,7])) # primary lifestyle levels

physeq@tax_table[,7] <- plyr::revalue(physeq@tax_table[,7],
                                      c("algal_parasite" = "other_lifestyle",
                                        "animal_parasite" = "other_lifestyle",
                                        "dung_saprotroph" = "saprotroph",
                                        "ectomycorrhizal" = "ectomycorrhizal",
                                        "lichen_parasite" = "other_lifestyle",
                                        "lichenized" = "other_lifestyle",
                                        "litter_saprotroph" = "saprotroph",
                                        "mycoparasite" = "other_lifestyle",
                                        "plant_pathogen" = "other_lifestyle",
                                        "root_endophyte" = "other_lifestyle",
                                        "soil_saprotroph" = "saprotroph",
                                        "sooty_mold" = "other_lifestyle",
                                        "unspecified_saprotroph" = "saprotroph",
                                        "wood_saprotroph" = "saprotroph"))
```

```{r subsets}
# subsetting based on sites and guilds 
# CTRL F + remove_empty_samples = T <-> remove_empty_samples = F

# guilds of whole dataset
ecm_phy <- subset_taxa(physeq, primary_lifestyle == "ectomycorrhizal") %>% 
clean_pq(remove_empty_samples = FALSE, remove_empty_taxa = T, clean_samples_names = FALSE)
sapr_phy <- subset_taxa(physeq, primary_lifestyle == "saprotroph") %>% 
clean_pq(remove_empty_samples = F,remove_empty_taxa = T, clean_samples_names = F)
other_phy <- subset_taxa(physeq, (primary_lifestyle == "other_lifestyle" | is.na(primary_lifestyle)))  %>% clean_pq(remove_empty_samples = F,remove_empty_taxa = T, clean_samples_names = F)

# by year
ecm_12 <- subset_samples(ecm_phy, ecm_phy@sam_data$year=="2012") %>% clean_pq(remove_empty_samples = FALSE,remove_empty_taxa = T, clean_samples_names = FALSE)
ecm_18 <- subset_samples(ecm_phy, ecm_phy@sam_data$year=="2018") %>% clean_pq(remove_empty_samples = FALSE,remove_empty_taxa = T, clean_samples_names = FALSE)

orient <- subset_samples(ecm_phy, ecm_phy@sam_data$specification_value %in% c("N", "S")) %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)
stream <- subset_samples(ecm_phy, ecm_phy@sam_data$specification_value %in% c("UP", "LO")) %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)

# by treatment
ecm_for <- subset_samples(ecm_phy, ecm_phy@sam_data$treatment_code == "FC") %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)

ecm_edg <- subset_samples(ecm_phy, ecm_phy@sam_data$treatment_code == "FE") %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)

ecm_cut <- subset_samples(ecm_phy, ecm_phy@sam_data$treatment_code == "SC") %>% clean_pq(remove_empty_samples = T,remove_empty_taxa = T, clean_samples_names = T)

fruit12 <- fruit[, grep("2012", colnames(fruit))]
fruit18 <- fruit[, grep("2018", colnames(fruit))]
# changing 0 to NAs for chi test
fruit12[rowSums(fruit12) == 0, ] <- NA
fruit18 <- fruit18[!rowSums(fruit18) == 0, ]

# theorical values
## 
fruit12_th <- ifelse(fruit12 >= 0, 33.33333, 0)
fruit18_th <- ifelse(fruit18 >= 0, 33.33333, 0)
```

*VAR TRANSFO FC ANALYSIS*
```{r transfo variables}
sam <- lapply(physeq@sam_data, as.factor) %>% as.data.frame()
sample_data(physeq)$year_factor <- as.factor(sample_data(physeq)$year)
```

```{r transposing otu table in physq object for hill_pq}
phy_otu_t <- t(physeq@otu_table)
phy_sam <- physeq@sam_data
phy_tax <- physeq@tax_table

ecm_phy_otu_t <- t(ecm_phy@otu_table) %>% otu_table()
ecm_phy_t <- phyloseq(ecm_phy_otu_t, ecm_phy@sam_data, ecm_phy@tax_table) %>% clean_pq()
```

```{r grouping by factor of insterest}
# grouping per level of treatment for raref curves
var1 <- as.character(get_variable(ecm_phy, "year"))
var2 <- as.character(get_variable(ecm_phy, "treatment_code"))
sample_data(ecm_phy)$yr_treat <- mapply(paste0, var1, var2, collapse = "_")
phy_merged <- merge_group(ecm_phy, "yr_treat")

yr_12 <- merge_group(ecm_12, group = "treatment_code",)
yr_18 <- merge_group(ecm_18, group = "treatment_code",)
```

```{r diversity indexes}
# creating diversity variables
div <- vegan::diversity(x = t(ecm_phy@otu_table),
                             index = "shannon")
rich <- vegan::specnumber(x= t(ecm_phy@otu_table))
even <- div/ log(rich) # Pielou's evenness # tend to 1 = more evenly distributed 
tax_f$ID_TYPE[is.na(tax_f$ID_TYPE)] <- "AUTO_ID" 
even[even == 0] <- even["NaN"]
df_div <- data.frame(ecm_phy@sam_data, div, rich, even) 
colnames(df_div)
df_div[, c(3:12)] <- lapply(df_div[, c(3:12)], as.factor) # tranforming columns into factors

df_div12 <- subset(df_div, year == "2012")
df_div18 <- subset(df_div, year == "2018")

stde_ <- tapply(df_div$div, df_div$treatment_code, stde) # if needed for tables
median_ <- tapply(df_div$div,df_div$treatment_code, median)
min_ <- tapply(df_div$div,df_div$treatment_code, min)
max_ <- tapply(df_div$div,df_div$treatment_code, max)

# for stream test
div_str <- vegan::diversity(x = t(stream@otu_table),
                             index = "shannon")
rich_str <- vegan::specnumber(x= t(stream@otu_table))

df_div_str <- data.frame(stream@sam_data, div_str, rich_str) 
colnames(df_div_str)
df_div_str[, c(3:12)] <- lapply(df_div_str[, c(3:12)], as.factor) # tranforming columns into factors


stde_str <- tapply(df_div_str$div, df_div_str$treatment_code, stde) # if needed for tables
median_str <- tapply(df_div_str$div,df_div_str$treatment_code, median)
min_str <- tapply(df_div_str$div,df_div_str$treatment_code, min)
max_str <- tapply(df_div_str$div,df_div_str$treatment_code, max)
```
```{r relative proportion of ECM}

rich_sapr <- vegan::specnumber(x = t(sapr_phy@otu_table))
rich_sapr_df <- data.frame(rich_sapr, c("SAPR"))
rich_ecm <- vegan::specnumber(x = t(ecm_phy@otu_table))
rich_emc_df <- data.frame(rich_ecm, c("ECM"))

# vertical join 
df_div_ecm <- data.frame(phy@sam_data, rich_ecm, c("ECM"))
colnames(df_div_ecm)
colnames(df_div_ecm)[13] <- "richness"
colnames(df_div_ecm)[14] <- "primary_lifestyle"
df_div_sapr <- data.frame(phy@sam_data, rich_sapr, c("SAPR"))
colnames(df_div_sapr)[13] <- "richness"
colnames(df_div_sapr)[14] <- "primary_lifestyle"

df_div_relat <- rbind(df_div_ecm, df_div_sapr)
df_div_relat[, c(3:12)] <- lapply(df_div_relat[, c(3:12)], as.factor) # tranforming columns into factors


## relative abundance
# Agglomerate taxa at phylum level
phy_relat <- tax_glom(physeq, taxrank = "primary_lifestyle")

# Calculate relative abundance
phy_relat1 <- transform_sample_counts(phy_relat, function(x) x / sum(x))

# Subset object to only ECM
phy_relat2 <- subset_taxa(phy_relat1, primary_lifestyle == "ectomycorrhizal")
ecm_abun <- colSums(phy_relat2@otu_table) # total sum of ECM relative abundance / sample
ecm_abun_df <- data.frame(phy_relat2@sam_data$sample_id, phy_relat2@sam_data$site, phy_relat2@sam_data$plot, phy_relat2@sam_data$year, phy_relat2@sam_data$treatment_code, ecm_abun)
colnames(ecm_abun_df) <- c("sample_id", "site", "plot", "year", "treatment_code", "ecm_abun")
ecm_abun_df[, c(2:5)] <- lapply(ecm_abun_df[, c(2:5)], as.factor) # tranforming columns into factors
ecm_abun_df$ecm_abun <- pmax(ecm_abun_df$ecm_abun, 0.001) # adding a small constant to every values to respect beta_family requirements

ecm_abun_df12 <- subset(ecm_abun_df, year == "2012")
ecm_abun_df18 <- subset(ecm_abun_df, year == "2018")
```

```{r genus count}
ecm_otusam <- data.frame(ecm_phy@sam_data, t(ecm_phy@otu_table))
ecm_otusam12 <- subset(ecm_otusam, ecm_otusam$year == "2012")
ecm_otusam18 <- subset(ecm_otusam, ecm_otusam$year == "2018")
colnames(ecm_otusam)
ecm_phy1 <- ecm_phy
ecm_phy1 <- as_binary_otu_table(ecm_phy1)
rownames(ecm_phy1@otu_table) <- ecm_phy1@tax_table[, "Genus"]
osam <- data.frame(ecm_phy1@sam_data, t(ecm_phy1@otu_table))

# count 
colnames(osam)

# count each occurence by group and year
osam_count <- osam[, c(5, 9 ,14:length(colnames(osam)))] %>% 
  group_by(treatment_code, year) %>% 
  summarise_all(sum)

# order df 
osam_count <- osam_count[
  with(osam_count, order(year, rev(treatment_code))),
] %>% as.data.frame()
rownames(osam_count) <- str_c(osam_count$treatment_code, osam_count$year, sep = "_")

# adding % 
## calc N
table12 <- table(ecm_12@sam_data$treatment_code, ecm_12@sam_data$plot) %>% as.data.frame.matrix()

table18 <- table(ecm_18@sam_data$treatment_code, ecm_18@sam_data$plot) %>% as.data.frame.matrix()
Ntreat12 <- rowSums(table12) %>% print() 
Ntreat18 <- rowSums(table18) %>% print()

Nt12_df <- data.frame(treatment_code = c("SC", "FE", "FC"),year = "2012",  N = Ntreat12)
Nt18_df <- data.frame(treatment_code = c("SC", "FE", "FC"), year = "2018", N = Ntreat18)
N_df <- rbind(Nt12_df, Nt18_df)
N_df <- N_df[
  with(N_df, order(year, rev(treatment_code))),
] %>% as.data.frame() # N is now order as the same as osam_count treatment + year

# we can merge column to osam_count_df 

# adding N for each group
osam_count_df <- data.frame(N = N_df$N, osam_count)

sum(col(osam_count_df))
# Define the list of column names to calculate proportions for
colnames(osam_count_df)
col_names_prop <- colnames(osam_count_df[, c(4: length(colnames(osam_count_df)))]) # Add all relevant column names here

# Apply the prop_fun() function to all columns and get a list of proportion vectors
prop_fun <- function(x) {
  x <- x / osam_count_df$N * 100
} # where:
## x = cell value; osam_count_df$N = the nb of samples for the corresponding row; * 100 = to get the proportion of genus occurrence by sample type. 
osam_count_df$N
prop_list <- lapply(osam_count_df[, col_names_prop], prop_fun)

# Convert the list of proportion vectors to a data frame
pourc_genusdf <- as.data.frame(prop_list)

# rename the columns of the data frame to match the original column names
rownames(pourc_genusdf) <- rownames(osam_count_df)
colnames(pourc_genusdf) <- col_names_prop
pourc_genusdf <- pourc_genusdf %>% round_df(1) # rounding

# remove 0 samples:
pourc_genusdf <- subset(pourc_genusdf[c(2:ncol(pourc_genusdf))], select = -which(colSums(pourc_genusdf[c(2:ncol(pourc_genusdf))]) == 0))

# subsetting per year if needed
pourc_genusdf12 <- subset(pourc_genusdf, grepl("2012", rownames(pourc_genusdf)))  
pourc_genusdf18<- subset(pourc_genusdf, grepl("2018", rownames(pourc_genusdf))) 

# remove year from rowname, and only keep the treatment
rownames(pourc_genusdf12) <- gsub("_2012", "", rownames(pourc_genusdf12))
rownames(pourc_genusdf18) <- gsub("_2018", "", rownames(pourc_genusdf18))
```