---
title: "PRELIMINARY_RESULTS"
output: html_document
date: "2023-03-08"
---
```{r KNITR global options: TO DO NOT INCLUDE}
#| include = F
 
knitr::opts_chunk$set(echo = F, include = F)
```
# *MEMOS*
controls: 
alt+o = close all 
shift+alt+o = open all

for ggplot: position = fill -> relative abundance 
# *REMAINS TO DO* 
- rarefactions pb 
- venn pb (all common, nothing specific)
- hill (or similar)
- deciding between circle (focus on genus distribution among treatments) or barplot (focusing on treatment differences)
- mds-nmds etc.
- permanova/adonis
- additional arguments (cf. transparency in others function inside circle_pq)
- commenting ##
- synthetic tab for MM 
- ID the missing samples (and create a design table)

# *0 LIBRARIES - FUNCTIONS - PALETTES* 
```{r libraries}
library(plyr)
library(pals)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(vegan)
```
```{r dummy graphs}
list_mm <- lsf.str("package:MiscMetabar") 
factor(list_mm)
## to see all functions from a package

# circles physeq
circle_allsite_genus <- circle_pq(ecm_phy, "site", taxa = "Phylum", add_nb_seq = F,
          grid_col = "darkgrey",
          row_col = c("darkred","#1F78B4","darkgreen"))

sample_names(ecm_phy)
```
```{r color palette}
pal.bands(alphabet, alphabet2, cols25, glasbey, kelly, polychrome, 
  stepped, tol, watlington,
  show.names=FALSE)

funky <- colorRampPalette(c("#A6CEE3","#1F78B4","#B2DF8A",
"#33A02C","#FB9A99","#E31A1C",
"#FDBF6F","#FF7F00","#CAB2D6",
"#6A3D9A","#FFFF99","#B15928"))
```
```{r functions}
outersect <- function(x, y) {
  sort(c(setdiff(x, y),
         setdiff(y, x)))
} # extract uncommon values between x - y 
```
# *1 MM RESULTS*
## IMPORTATION DATA
```{r data import and some modifs}
# tax-eco table
setwd("H:/REDACTION/RESULTS/cutting_ecm/TRANSF_DATA2") # to change, depending on which

#data run
run_data <- read.table("run_data.txt",header=T,sep="\t",na.strings=T,dec=".")
colnames(run_data)

# clust all
raw_clust <- read.table("raw_clusters.txt",header=T,sep="\t",na.strings=T,dec=".") %>% 
arrange(desc(Cluster_Size))
raw_clust[raw_clust==""] <- NA
raw_clust[,4] <- as.character(raw_clust[,4])

raw_f_clust <- subset(raw_clust,!grepl("PLANT|NF",raw_clust[,4]))
ecm_raw <- subset(taxeco,taxeco$primary_lifestyle=="ectomycorrhizal") %>% arrange(desc(Cluster_Size))
sapr_raw <- subset(taxeco,taxeco$primary_lifestyle!="ectomycorrhizal") %>% arrange(desc(Cluster_Size))
```
## DATA RUN
```{r results mm}
colnames(run_data) 

run_data[,1]# 2 runs
run_data[,25] # 98,5% similarity clustering

# run 1
run_data[1,2] # total nb of pyro reads
run_data[1,3] # sequences passing quality control (QC)
run_data[1,2]-run_data[1,3] # sequences deleted

# run 2
run_data[2,2] # total nb of pyro reads
run_data[2,3] # sequences passing quality control (QC)
run_data[2,2]-run_data[2,3] # sequences deleted

# scata taxa attribution 

## no IDed sequences
is.na(raw_clust[,4]) %>% summary() # 1145 clusters ID / 4050 
pourc_id <- 1145/4050*100; pourc_id # autoID (especially the most abundant clusters)
# ID for 1000s first TODO

## NF sequences
grep("PLANT|NF", raw_clust[,4], value = TRUE) %>% length() # 82 non-fungi clusters IDed 

# total VS conserve
## nb clusters
nrow(raw_clust) # all
nrow(ecm_phy@tax_table) # ecm among 1000st more abundant Fungi clusters 
pourc_ecm <- nrow(ecm_phy@tax_table)/1000*100; pourc_ecm # 20% ECM guild among Fungi

## nb reads
run_data[3,2]
raw_total_read <- sum(raw_f_clust$Cluster_Size); raw_total_read ## TODO: exlcude plants to have proper pourcentages values
ecm_total_read <-sum(ecm_raw$Cluster_Size); ecm_total_read
pourc_nb_read_ecm <- ecm_total_read/raw_total_read*100; pourc_nb_read_ecm # pourc nb read conserved/ nb read 
summary(ecm_raw$Cluster_Size) # min, mean, max of ecm reads
```
## OVERVIEW_DATA
```{r general}
summary(run_data)
summary(raw_clust)
summary(ecm_raw)
summary(sapr_raw)
levels(droplevels(ecm_raw$Genus))
levels(droplevels(sapr_raw$Genus))

summary(ecm_raw$Cluster_Size)

hist_cluster_size <- ggplot(ecm_raw, 
       aes(x=Cluster_Size))+
geom_histogram(bins=20,breaks=seq(12,2985 ,by=100),color="black", fill="darkseagreen")+
  scale_y_continuous(name="number of clusters")+
  scale_x_continuous(name="number of reads", breaks=seq(12,2985 ,by=100))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=60))+
  ggtitle("Distribution of the cluster size for ECM OTUs") 

hist_cluster_size

read_distribution <- plot_read_distribution(ecm_phy, groups = "site", 
                            plot.type = "density")
print(read_distribution + theme_biome_utils())
```
```{r SAC nb read}
sac_f_raw <- ggplot(raw_f_clust, aes(x=c(1:3968),y=cumsum(Cluster_Size)))+   geom_line(color="red")+
  geom_point(size=0.7,alpha=0.5)+
  coord_cartesian(xlim=c(0,3968),ylim=c(0,244367))+
  scale_y_continuous(name="cumulative cluster size", breaks=seq(0,244367,by=25000))+
  scale_x_continuous(name="number of cluster", breaks=seq(0,3968 ,by=200))+
  geom_vline(xintercept=1000,lwd=0.6,colour="red",)+ 
  annotate("text", x = 1050, y = 25000, label="1000 first clusters conserved", size = 4, angle=0, color = "red", hjust = 0)+
  annotate("text", x= 1050, y= 15000, label="93.65% of total reads", size = 4, angle=0, color = "red", hjust = 0)+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=60))+
  ggtitle("Cumulative cluster size curve from all fungi clusters") 

sac_f_raw

read_1000 <- sum(raw_f_clust[c(1:1000),2]); read_1000
read_3968 <- sum(raw_f_clust[c(1001:3968),2]); read_3968
read_total <- sum(raw_f_clust[,2])
pourc_read_kept <- read_1000/read_total*100; pourc_read_kept
```
```{r sample desing}

nrow(otu) # total number of all clusters (ie fungi ecm or no-ecm, plants)
colnames(otu)

# samples missing post scata (ie not present in the OTU matrix from clustering = no match between ITS4 - ITS7 tags)

samples_missing_postscata <- outersect(x = sam_no_match$samples_match, y = sam_no_match$samples_all) # antijoin but for columns : extract samples not in common (missing from initial samples design)
samples_missing_postscata <- dput(as.character(samples_missing_postscata)) # convert into the format c(x; y; ...; z)
sam[sam$sample_id %in% samples_missing_postscata,] # showing samples with site and plots information missing 
## 5 controls; 3 Halasen; 2 Halasen_p5, 1 Q1; 6 PCR_neg
## total missing = 17

sample_count_design <- length(rownames(sam)) # 178 samples total design + controls)

# design samples per site 
length(rownames(sam_hala))
length(rownames(sam_hala5))
length(rownames(sam_fanga))

# sample count
length(rownames(sam_controls_na)) # 19 (12 NAs + 7 controls)
sum(is.na(sam_controls_na$site)) 
real_sample_count <- (length(rownames(sam))-length(rownames(sam_controls_na))) %>%
  print() # 159 sites samples


colnames(otu)
s_id_controls <- sam_controls_na$sample_id[sam_controls_na$site == "control"] %>%
  na.exclude()  # what are the samples_id of controls samples ? 
s_id_controls <- dput(as.character(s_id_controls))

col_control_kept <- which(colnames(otu) %in% s_id_controls, arr.ind = F) 
colnames(otu)[col_control_kept]

controls_kept <- c("sITS-standard2", "sITS-standard1", "sPCR_neg_1", "sPCR_neg_2", "sQ2", "s49", "s97") # 2 ITS; 2 PCR; 1Q; 2 controls 

161-7+5 
# 161 = total samples kept after scata
# -7  = controls kept after scata
# +5  = site samples missing 
## final 159 samples -> ALL GOOD 
```
```{r missing samples}
# considering physeq objects (ecm_phy,hala_ecm, hala5_ecm, fanga_ecm have been cleaned (sample supress if 0) by clean_pq() )
colnames(hala_sam)
summary(hala_sam[,c(2,5:8)])
table(hala_sam$plot,hala_sam$treatment)

# for ecm-sapr
summary(sam$sample_id)
summary(physeq@sam_data$sample_id)
miss_sam_clust_ecmsapr <- anti_join(x = sam, y = physeq@sam_data) # return all rows from x without a match in y.
miss_sam_clust_ecmsapr$sample_id # samples deleted after bioinformatic processes (scata, first 1000s clusters, ecm-sapr guilds)

# for ecm only
summary(sam$sample_id)
summary(ecm_phy@sam_data$sample_id)
miss_sam_clust_ecm <- anti_join(x = sam, y = ecm_phy@sam_data)
miss_sam_clust_ecm$sample_id
# 8 PCR neg => good control,
# what about ITS standard ?

clipr::write_clip(miss_sam_clust_ecm$sample_id) # to copy dataframe or vector
# controls and na 
miss_sam_clust_ctrl <- anti_join(x = sam_controls_na, y = ecm_phy@sam_data)
miss_sam_clust_ctrl$sample_id

length(miss_sam_clust_ctrl$sample_id)
length(row.names(sam_controls_na)) # every control and NA are suppressed from the dataset (because 0 occurence of clusters)

# hala
miss_sam_clust_hala <- anti_join(x = sam_hala[,1:7], y = hala_ecm@sam_data[,1:7]) # exclude specification column because we changed factor values for this column
miss_sam_clust_hala$sample_id

# hala5
miss_sam_clust_hala5 <- anti_join(x = sam_hala5, y = hala5_ecm@sam_data)
miss_sam_clust_hala5$sample_id

# fanga
miss_sam_clust_fanga <- anti_join(x = sam_fanga, y = fanga_ecm@sam_data)
miss_sam_clust_fanga$sample_id

# check if count is good (total missing samples VS sum of missing samples per site)
length(miss_sam_clust_ecm$sample_id)/(length(miss_sam_clust_hala$sample_id)+length(miss_sam_clust_hala5$sample_id)+length(miss_sam_clust_fanga$sample_id)+length(miss_sam_clust_ctrl$sample_id)) 
# =1 -> all good 

sam[sam$sample_id %in% samples_missing_postscata,] # showing samples with site and plots information missing 
## 5 controls; 3 Halasen; 2 Halasen_p5, 1 Q1; 6 PCR_neg
## total missing = 17
length(miss_sam_clust_ecm$sample_id) # total 
length(miss_sam_clust_ctrl$sample_id)

length(miss_sam_clust_hala$sample_id) # 3 missing post scata; 3 missing post taxa-attrib
length(miss_sam_clust_hala5$sample_id)# 2 missing post scata; 2 missing post taxa-attrib
length(miss_sam_clust_fanga$sample_id)# 0 missing post scata; 20 missing post taxa-attrib

length(rownames(sam_hala))
length(rownames(sam_hala5))
length(rownames(sam_fanga))
```
```{r potential contamination ?}
which(samples_kept_pq>0, arr.ind = T) # location where values of controls have reads
clusters_reads_inctrl <- rownames(which(samples_kept_pq>0, arr.ind = T)) %>% as.factor()
levels(clusters_reads_inctrl) # clusters where we found reads in controls samples 
## -> should we erase them for dataset ? 
## surely, imo 
```
# *2 OVERVIEW DATA FROM SITES*
```{r between sites}

ggvenn_pq(ecm_phy, fact="site", taxonomic_rank = "Genus") 
# not working well

circle_pq(ecm_phy, "site", taxa = "Genus", add_nb_seq = F) # only 53,57% plotted
circle_pq(ecm_phy, "site", taxa = "Phylum", add_nb_seq = F,
          grid_col = "darkgrey",
          row_col = c("darkred","#1F78B4","darkgreen"))

read_distrib <- plot_read_distribution(
  ecm_phy, groups = "site", 
  plot.type = "density")+
  xlab("Reads per site")+
  ggtitle("Distribution of the number of reads per site")

# rarecurve by samplings
ecm_phy_rrcurve <- phyloseq.extended::ggrare(physeq = ecm_phy, 
                                             step = 10, 
                                             color = "year", 
                                             label="sample_id",
                                             plot=F, 
                                             se=T)+ 
  facet_wrap(~site)+
  ggtitle("Rarefaction curves of samples per site")+
  xlab("Number of reads")+
  ylab("OTU richness")

# rarecurve by groups (and confidence interval)
ecm_phy_rrcurve_group <- microbiomeutilities::plot_alpha_rcurve(ecm_phy, index="observed", 
                       subsamples = seq(2,2000,by=50),
                       lower.conf = , 
                       upper.conf = ,
                       group="site",
                       label.color = "brown3",
                       label.size = 3,
                       label.min = TRUE,
                       type = "CI") 
ecm_phy_rrcurve_group
```
# *3 HALASEN STUDY: edge retention trees*
## 3.1 design and sampling
```{r overview }
# sampling design
colnames(hala_sam)
summary(hala_sam[,c(2,5:8)])
table(hala_sam$plot,hala_sam$treatment)

# nb genus
colnames(hala_ecm@tax_table)
as.factor(hala_ecm@tax_table[,5]) %>% droplevels() %>% levels() 
genus_nb_hala <- as.factor(hala_ecm@tax_table[,5]) %>% droplevels() %>% levels() %>% length()
## 23 genus
## attribution facolors, or whatever to have proper constrasted colors on graphicals representations

ggvenn_pq(hala_ecm, fact="year", taxonomic_rank = "Genus")
```

- change between 2012 2018 (see below) -> could be explained by a difference in the biomolecular methods ? we should not expect any difference in forest control treatment between years

## 3.2.1 alpha
```{r rarecurve}
hala_rrcurve_groups1 <- phyloseq.extended::ggrare(hala_grp_12, step = 10, color = "specification",plot=F, se=T)+ 
  facet_wrap(~year)+
  labs(x = "Group size",
       y =  "OTU richness",
       title = "rarefaction curves by treatment",
       color = "Treatment")+
  guides(fill = F)+ # Suppressing the filling (standard error) legend 
  scale_color_manual(values = c("CC_edge" = "orange",
                                "CC_middle" = "red",
                                "FOR_middle" = "darkgreen"))+
  scale_fill_manual(values = c("CC_edge" = "orange",
                                "CC_middle" = "red",
                                "FOR_middle" = "darkgreen"))
hala_rrcurve_groups1

hala_rrcurve_groups2 <- phyloseq.extended::ggrare(hala_grp_18, step = 10, color = "specification",plot=F, se=T)+ 
  facet_wrap(~year)+
  labs(x = "Group size",
       y =  "OTU richness",
       title = "rarefaction curves by treatment",
       color = "Treatment")+
  guides(fill = F)+ # Suppressing the filling (standard error) legend 
  scale_color_manual(values = c("CC_edge" = "orange",
                                "CC_middle" = "red",
                                "FOR_middle" = "darkgreen"))+
  scale_fill_manual(values = c("CC_edge" = "orange",
                                "CC_middle" = "red",
                                "FOR_middle" = "darkgreen"))
hala_rrcurve_groups2
```
```{r circle graph: genus distribution among treatments}
circle_pq(physeq = hala_ecm, 
                              fact = "specification", 
                              taxa = "Genus", 
                              min_prop_mod=0.001,
                              min_prop_tax=0.01,
                              row_col = c("CC_middle" = "red",
                                          "CC_edge" = "orange",
                                          "FOR_middle" = "darkgreen"),
                              grid_col= 1,
                              add_nb_seq = T)
```
```{r hill otu}
hala_hill_treatment <- hill_pq(physeq = hala_ecm_t, 
                               variable = "specification",
                               color_fac = NA, 
                               letters = F)
  

hala_hill0 <- hala_hill_treatment[[1]]+ 
  labs(title="Hill 0 : position from CC", subtitle="Hagadalen_edge")+
  coord_cartesian(xlim=c(0,16))+
  scale_x_continuous(name="Hill_0", breaks=seq(0,16,by=2))+
  scale_y_continuous(name="Position from CC", breaks=c(1,2,3))+
  scale_color_manual(values = c("CC_edge" = "orange",
                                "CC_middle" = "red",
                                "FOR_middle" = "darkgreen"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))
hala_hill0

hala_hill1 <- hala_hill_treatment[[2]]+ 
  labs(title="Hill 1 : position from CC", subtitle="Hagadalen_edge")+
  coord_cartesian(xlim=c(0,12))+
  scale_x_continuous(name="Hill_1", breaks=seq(0,12,by=2))+
  scale_y_continuous(name="Position from CC", breaks=c(1,2,3))+
  scale_color_manual(values = c("CC_edge" = "orange",
                                "CC_middle" = "red",
                                "FOR_middle" = "darkgreen"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))
hala_hill1

hala_hill2 <- hala_hill_treatment[[3]]+ 
  labs(title="Hill 2 : position from CC", subtitle="Hagadalen_edge")+
  coord_cartesian(xlim=c(0,10))+
  scale_x_continuous(name="Hill_2", breaks=seq(0,10,by=2))+
  scale_y_continuous(name="Position from CC", breaks=c(1,2,3))+
  scale_color_manual(values = c("CC_edge" = "orange",
                                "CC_middle" = "red",
                                "FOR_middle" = "darkgreen"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))
hala_hill2

hala_hill_treatment[[4]]
hala_hill_treatment[[4]] # TukeyHSD test
```
```{r barplot genus distribution}
genus_nb_hala <- as.factor(hala_ecm@tax_table[,5]) %>% droplevels() %>% levels() %>% length()
## OTU Genus count (0;1) by treatment
hala_ecm_ggplot1 <- phyloseq::plot_bar(as_binary_otu_table(hala_ecm), x = "specification") 
hala_ecm_ggplot1 <- ggplot(hala_ecm_ggplot1$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=specification, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  scale_fill_manual(values = funky(genus_nb_hala))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus count (0;1) by position from CC and year", subtitle = "Halasen_edge")+
  xlab("position from CC")+
  guides(fill=guide_legend(title="Genus"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=20))
hala_ecm_ggplot1

# OTU Genus abundance by treatment
hala_ecm_ggplot2 <- phyloseq::plot_bar(hala_ecm, x = "specification") 
hala_ecm_ggplot2 <- ggplot(hala_ecm_ggplot2$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=specification, y = Abundance))+
  geom_bar(stat="identity", position = "stack")+
  scale_fill_manual(values = funky(genus_nb_hala))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus abundance by position from CC and year", subtitle = "Halasen_edge")+
  xlab("position from CC")+
  guides(fill=guide_legend(title="Genus"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=20))
hala_ecm_ggplot2
```
```{r barplot OTUs distribution}
otu_nb_hala <- as.factor(hala_ecm@tax_table[,6]) %>% droplevels() %>% levels() %>% length()
## OTU count (0;1) by treatment
hala_ecm_ggplot3 <- phyloseq::plot_bar(as_binary_otu_table(hala_ecm), x = "specification") 
hala_ecm_ggplot3 <- ggplot(hala_ecm_ggplot3$data, aes(fill= fct_reorder(as.factor(OTU_ID), Abundance), x=specification, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  scale_fill_manual(values = funky(otu_nb_hala))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus count (0;1) by position from CC and year", subtitle = "Halasen_edge")+
  xlab("position from CC")+
  guides(fill=guide_legend(title="Genus"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=20))
hala_ecm_ggplot3
```
## 3.2.2 multivariate analysis
```{r multivariate representation}

hala_ecm_ordi_mdsjacc <- ordinate(as_binary_otu_table(hala_ecm), method = "MDS", dist="jaccard")
hala_ecm_ordi_mdsjacc <- plot_ordination(hala_ecm, hala_ecm_ordi_mdsjacc, type="samples", color="specification",label="sample_id")+
  geom_point(size=2)+
  facet_wrap(~year)+
  # color and elipse
  scale_color_manual(values = c("CC_edge" = "orange",
                              "CC_middle" = "red",
                              "FOR_middle" = "darkgreen"))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="position from CC"))+
  ggtitle(label = "MDS of samples from Halasen_edge by year", subtitle="presence-absence (jaccard distance)")
hala_ecm_ordi_mdsjacc

hala_ecm_ordi_mdsbray <- ordinate(hala_ecm, method = "MDS", dist="bray")
hala_ecm_ordi_mdsbray <- plot_ordination(hala_ecm, hala_ecm_ordi_mdsbray, type="samples", color="specification",label="sample_id")+
  geom_point(size=2)+
  facet_wrap(~year)+
  # color and elipse
  scale_color_manual(values = c("CC_edge" = "orange",
                              "CC_middle" = "red",
                              "FOR_middle" = "darkgreen"))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="position from CC"))+
  ggtitle(label = "MDS of samples from Halasen_edge by year",subtitle = "abundance (bray-curtis distance)")
hala_ecm_ordi_mdsbray

hala_ecm_ordi_phyl_mdsbray <- ordinate(hala_ecm, method = "MDS", dist="bray")
plot_ordination(hala_ecm, hala_ecm_ordi_phyl_mdsbray, type="taxa", color="Order", shape="Phylum",label="sample_id")+
  geom_point(size=2)+
  # color and elipse
  scale_color_manual(values =cols25(10))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="position from CC"))+
  ggtitle(label = "MDS of samples from Halasen_edge by year",subtitle = "abundance (bray-curtis distance)")
```
# *4 HALASEN_5P STUDY: randomly distributed retention trees*
## 4.1 design and sampling
```{r overview }
colnames(hala5_sam)
summary(hala5_sam[,c(2,5:8)])
summary(hala5_sam$plot)
table(hala5_sam$plot,hala5_sam$treatment)
print(hala5_ecm@sam_data)

# nb genus
colnames(hala5_ecm@tax_table)
as.factor(hala5_ecm@tax_table[,5]) %>% droplevels() %>% levels()
genus_nb_hala5 <- as.factor(hala5_ecm@tax_table[,5]) %>% droplevels() %>% levels() %>% length() 
genus_nb_hala5 # 18 genus
```
## 4.2.1 bivariate analysis
```{r rarecurve}
hala5_rrcurve0 <- phyloseq.extended::ggrare(hala5_grp, step = 10, color = "treatment",plot=F, se=T)+
  labs(x = "Group size",
       y =  "OTU richness",
       title = "rarefaction curves by treatment",
       color = "Treatment")+
  guides(fill = F)+ # Suppressing the filling (standard error) legend 
  scale_color_manual(values = c("forest" = "darkgreen",
                                "thinned" = "orange",
                                "clearcut" = "red"))+
  scale_fill_manual(values = c("forest" = "darkgreen",
                                "thinned" = "orange",
                                "clearcut" = "red"))
hala5_rrcurve0
```
```{r circle graph: genus distribution among treatments}
circle_pq(physeq = hala5_ecm, 
                              fact = "treatment", 
                              taxa = "Genus", 
                              min_prop_mod=0.001,
                              min_prop_tax=0.01,
                              row_col = c("clearcut" = "red",
                                          "forest" ="darkgreen",
                                          "thinned" = "orange"),
                              grid_col= 1,
                              add_nb_seq = T) # distribution circle sample - taxonomy 
```
```{r genus distribution}
genus_nb_hala5 <- as.factor(hala5_ecm@tax_table[,5]) %>% droplevels() %>% levels() %>% length()
hala5_ecm_ggplot1 <- phyloseq::plot_bar(as_binary_otu_table(hala5_ecm), x = "treatment") 
hala5_ecm_ggplot1 <- ggplot(hala5_ecm_ggplot1$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  scale_fill_manual(values = cols25(genus_nb_hala5))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus count (0;1) by cutting treatment",subtitle = "Halasen_P5")+
  guides(fill=guide_legend(title="Genus"))
hala5_ecm_ggplot1

hala5_ecm_ggplot2 <- phyloseq::plot_bar(hala5_ecm, x = "treatment") 
hala5_ecm_ggplot2 <- ggplot(hala5_ecm_ggplot2$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  scale_fill_manual(values = cols25(genus_nb_hala5))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus abundance (nb of reads) by cutting treatment",subtitle = "Halasen_P5")+
  guides(fill=guide_legend(title="Genus"))
hala5_ecm_ggplot2
```
```{r OTUs distribution}
otu_nb_hala5 <- as.factor(hala5_ecm@tax_table[,6]) %>% droplevels() %>% levels() %>% length()
hala5_ecm_ggplot3 <- phyloseq::plot_bar(as_binary_otu_table(hala5_ecm), x = "treatment")
hala5_ecm_ggplot3 <- ggplot(hala5_ecm_ggplot3$data, aes(fill= fct_reorder(as.factor(OTU_ID), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  scale_fill_manual(values = funky(otu_nb_hala5))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus count (0;1) by cutting treatment",subtitle = "Halasen_P5")+
  guides(fill=guide_legend(title="Genus"))
hala5_ecm_ggplot3

hala5_ecm_ggplot4 <- phyloseq::plot_bar(hala5_ecm, x = "treatment") 
hala5_ecm_ggplot4 <- ggplot(hala5_ecm_ggplot4$data, aes(fill= fct_reorder(as.factor(OTU_ID), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  scale_fill_manual(values = funky(otu_nb_hala5))+
  facet_grid(~year, scale="free")+
  labs(title="OTU abundance (nb of reads) by cutting treatment",subtitle = "Halasen_P5")+
  guides(fill=guide_legend(title="Genus"))
hala5_ecm_ggplot4
```
```{r hill OTUs}
hala5_hill_treatment <- hill_pq(physeq = hala5_ecm_t, variable = "treatment", color_fac = NA, letters = F) 
hala5_hill0 <- hala5_hill_treatment[[1]]+ 
  labs(title="Hill 0 : apha diversity of ", subtitle="Halasen5")+
  coord_cartesian(xlim=c(0,16))+
  scale_x_continuous(name="Hill_0", breaks=seq(0,16,by=2))+
  scale_y_continuous(name="treatment", breaks=c(1,2,3))+
  scale_color_manual(values = c("forest" = "darkgreen",
                                "thinned" = "orange",
                                "clearcut" = "red"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))
hala5_hill0

hala5_hill1 <- hala5_hill_treatment[[2]]+ 
  labs(title="Hill 1 : treatment", subtitle="Halasen5")+
  coord_cartesian(xlim=c(0,12))+
  scale_x_continuous(name="Hill_1", breaks=seq(0,12,by=2))+
  scale_y_continuous(name="treatment", breaks=c(1,2,3))+
  scale_color_manual(values = c("forest" = "darkgreen",
                                "thinned" = "orange",
                                "clearcut" = "red"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))

hala5_hill2 <- hala5_hill_treatment[[3]]+ 
  labs(title="Hill 2 : treatment", subtitle="Halasen5")+
  coord_cartesian(xlim=c(0,10))+
  scale_x_continuous(name="Hill_2", breaks=seq(0,10,by=2))+
  scale_y_continuous(name="treatment", breaks=c(1,2,3))+
  scale_color_manual(values = c("forest" = "darkgreen",
                                "thinned" = "orange",
                                "clearcut" = "red"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))

hala5_hill_treatment[[4]]
hala5_hill_treatment[[4]] # TukeyHSD test
```

## 4.2.2 multivariate analysis
```{r ordination}
hala5_ecm_ordi_mdsjacc  <- ordinate(hala5_ecm, method = "MDS", dist="bray")
hala5_ecm_ordi_mdsjacc  <- plot_ordination(hala5_ecm, hala5_ecm_ordi_mdsjacc, type="samples", color="treatment",label="sample_id")+
  geom_point(size=2)+
  # color and elipse
  scale_color_manual(values = c("clearcut" = "red",
                                "thinned" = "orange",
                                "forest" = "darkgreen"))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="treatment"))+
  ggtitle(label = "MDS of samples from Halasen_5P",subtitle = "abundance")
hala5_ecm_ordi_mdsjacc

hala5_ecm_ordi_mdsbray  <- ordinate(as_binary_otu_table(hala5_ecm), method = "MDS", dist="jaccard")
hala5_ecm_ordi_mdsbray  <-plot_ordination(hala5_ecm, hala5_ecm_ordi_mdsbray , type="samples", color="treatment",label="sample_id")+
  geom_point(size=2)+
  # color and elipse
    scale_color_manual(values = c("clearcut" = "red",
                                "thinned" = "orange",
                                "forest" = "darkgreen"))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="treatment"))+
  ggtitle(label = "MDS of samples from Halasen_5P",subtitle = "abundance")

hala5_ecm_ordi1 <- ordinate(as_binary_otu_table(hala5_ecm), method = "NMDS", dist="jaccard",k=20)
plot_ordination(hala5_ecm, hala5_ecm_ordi1, type="samples", color="treatment",label="sample_id")+
  geom_point(size=2)+
  # color and elipse
    scale_color_manual(values = c("clearcut" = "red",
                                "thinned" = "orange",
                                "forest" = "darkgreen"))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="treatment"))+
  ggtitle(label = "MDS of samples from Halasen_5P",subtitle = "abundance")

hala5_ecm_ordi_phyl1 <- ordinate(hala5_ecm, method = "MDS", dist="bray")
hala5_ecm_ordi_phyl1 <- plot_ordination(hala5_ecm, hala5_ecm_ordi_phyl1, type="taxa", color="Order", shape="Phylum")+
  geom_point(size=2)+
  # color and elipse
  scale_color_manual(values =cols25(10))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="position from CC"))+
  ggtitle(label = "MDS of samples from Halasen_edge by year",subtitle = "abundance (bray-curtis distance)")
hala5_ecm_ordi_phyl1
```
# *5 FANGAMON STUDY: randomly distributed retention trees*
## 5.1 design and sampling
```{r overview }
colnames(fanga_sam)
summary(fanga_sam[,c(2,5:8)])
summary(fanga_sam$plot)
table(fanga_sam$plot,fanga_sam$treatment)

# nb genus
colnames(fanga_ecm@tax_table)
as.factor(fanga_ecm@tax_table[,5]) %>% droplevels() %>% levels()
genus_nb_fanga <- as.factor(fanga_ecm@tax_table[,5]) %>% droplevels() %>% levels() %>% length()
genus_nb_fanga # 15 genus

```
## 5.2.1 bivariate analysis
```{r rarecurve}
fanga_rrcurve0 <- phyloseq.extended::ggrare(fanga_grp_12, step = 10, color = "treatment",plot=F, se=T)+ facet_wrap(~year)+
  labs(x = "Group size",
       y =  "OTU richness",
       title = "rarefaction curves by treatment",
       color = "Treatment")+
  guides(fill = F)+ # Suppressing the filling (standard error) legend 
  scale_color_manual(values = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"))+
  scale_fill_manual(values = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"))
fanga_rrcurve0

fanga_rrcurve1 <- phyloseq.extended::ggrare(fanga_grp_18, step = 10, color = "treatment",plot=F, se=T)+ facet_wrap(~year)+
  labs(x = "Group size",
       y =  "OTU richness",
       title = "rarefaction curves by treatment",
       color = "Treatment")+
  guides(fill = F)+ # Suppressing the filling (standard error) legend 
  scale_color_manual(values = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"))+
  scale_fill_manual(values = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"))
fanga_rrcurve1
```
```{r circle graph: genus distribution among treatments}
circle_fanga_ecm <- circle_pq(physeq = fanga_ecm, 
                              fact = "treatment", 
                              taxa = "Genus", 
                              min_prop_mod=0.001,
                              min_prop_tax=0.02,
                              row_col = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"),
                              grid_col= 1,
                              add_nb_seq = T)
```
```{r genus distribution}
genus_nb_fanga <- as.factor(fanga_ecm@tax_table[,5]) %>% droplevels() %>% levels() %>% length()
fanga_ecm_ggplot1 <- phyloseq::plot_bar(as_binary_otu_table(fanga_ecm), x = "treatment") 
fanga_ecm_ggplot1 <- ggplot(fanga_ecm_ggplot1$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  facet_grid(~year, scale="free")+
  scale_fill_manual(values = cols25(genus_nb_fanga))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus count (0;1) by cutting treatment",subtitle = "Fangamon")+
  guides(fill=guide_legend(title="Genus"))
fanga_ecm_ggplot1

fanga_ecm_ggplot2 <- phyloseq::plot_bar(fanga_ecm, x = "treatment") 
fanga_ecm_ggplot2 <- ggplot(fanga_ecm_ggplot2$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  facet_grid(~year, scale="free")+
  scale_fill_manual(values = cols25(genus_nb_fanga))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus abundance (nb of reads) by cutting treatment",subtitle = "Fangamon")+
  guides(fill=guide_legend(title="Genus"))
fanga_ecm_ggplot2
# position = fill -> relative abundance 
```
```{r OTU distribution}
otu_nb_fanga <- as.factor(fanga_ecm@tax_table[,6]) %>% droplevels() %>% levels() %>% length()
fanga_ecm_ggplot3 <- phyloseq::plot_bar(as_binary_otu_table(fanga_ecm), x = "treatment") 
fanga_ecm_ggplot3 <- ggplot(fanga_ecm_ggplot3$data, aes(fill= fct_reorder(as.factor(OTU_ID), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  facet_grid(~year, scale="free")+
  scale_fill_manual(values = funky(otu_nb_fanga))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus count (0;1) by cutting treatment",subtitle = "Fangamon")+
  guides(fill=guide_legend(title="Genus"))
fanga_ecm_ggplot3

fanga_ecm_ggplot4 <- phyloseq::plot_bar(fanga_ecm, x = "treatment") 
fanga_ecm_ggplot4 <- ggplot(fanga_ecm_ggplot4$data, aes(fill= fct_reorder(as.factor(OTU_ID), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  facet_grid(~year, scale="free")+
  scale_fill_manual(values = funky(otu_nb_fanga))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus abundance (nb of reads) by cutting treatment",subtitle = "Fangamon")+
  guides(fill=guide_legend(title="Genus"))
fanga_ecm_ggplot4
# position = fill -> relative abundance 
```
```{r hill fanga}
fanga_hill_treatment <- hill_pq(physeq = fanga_ecm_t,
                                variable = "treatment", 
                                color_fac = NA, 
                                letters = F) 

fanga_hill0 <- fanga_hill_treatment[[1]]+ 
  labs(title="Hill 0 : treatment", subtitle="Fangamon")+
  coord_cartesian(xlim=c(0,16))+
  scale_x_continuous(name="Hill_0", breaks=seq(0,16,by=2))+
  scale_y_continuous(name="treatment", breaks=c(1,2,3))+
  scale_color_manual(values = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))
fanga_hill0

fanga_hill1 <- fanga_hill_treatment[[2]]+ 
  labs(title="Hill 1 : treatment", subtitle="Fangamon")+
  coord_cartesian(xlim=c(0,12))+
  scale_x_continuous(name="Hill_1", breaks=seq(0,12,by=2))+
  scale_y_continuous(name="treatment", breaks=c(1,2,3))+
  scale_color_manual(values = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))

fanga_hill2 <- fanga_hill_treatment[[3]]+ 
  labs(title="Hill 2 : treatment", subtitle="Fangamon")+
  coord_cartesian(xlim=c(0,10))+
  scale_x_continuous(name="Hill_2", breaks=seq(0,10,by=2))+
  scale_y_continuous(name="treatment", breaks=c(1,2,3))+  
  scale_color_manual(values = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"))+
  theme(axis.text.x=element_text(color="coral4", size=8, angle=0))

fanga_hill_treatment[[4]]
fanga_hill_treatment[[4]] # TukeyHSD test
```
```{r trying understand difference between boolean and relative abundance}
fanga_ecm_only_cc <- subset_samples(fanga_ecm, fanga_ecm@sam_data$treatment=="clearcut") %>% clean_physeq(remove_empty_samples = T,remove_empty_taxa = T)

print(fanga_ecm_only_cc@otu_table)
which(fanga_ecm_only_cc@otu_table>200,arr.ind = T) %>% rownames() 
fanga_ecm_only_cc@tax_table[c("c10", "c21"),6] # 2 super abundant species 
```
```{r maybe a contamination for a sample?}
fanga_ecm_only_cc_2012 <- subset_samples(fanga_ecm, fanga_ecm@sam_data$year=="2012" & fanga_ecm@sam_data$treatment=="clearcut") %>% clean_physeq(remove_empty_samples = T,remove_empty_taxa = T)
print(fanga_ecm_only_cc_2012@sam_data)
print(fanga_ecm_only_cc_2012@otu_table) # c10 in both samples, c21 in another different one... no specific contamination it seems

fanga_ecm_excl_samples <- subset_samples(fanga_ecm, !fanga_ecm@sam_data$sample_id %in% c("s152","s45")) %>% clean_physeq(remove_empty_samples = T,remove_empty_taxa = T)

fanga_ecm_ggplot3 <- phyloseq::plot_bar(as_binary_otu_table(fanga_ecm_excl_samples), x = "treatment") 
ggplot(fanga_ecm_ggplot3$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  facet_grid(~year, scale="free")+
  scale_fill_manual(values = cols25(genus_nb_fanga))+
  facet_grid(~year, scale="free")+
  labs(title="OTU Genus count (0;1) by cutting treatment",subtitle = "Fangamon- exlcuded s152,s45 samples")+
  guides(fill=guide_legend(title="Genus"))

fanga_ecm_ggplot2 <- phyloseq::plot_bar(fanga_ecm_excl_samples, x = "treatment") 
ggplot(fanga_ecm_ggplot2$data, aes(fill= fct_reorder(as.factor(Genus), Abundance), x=treatment, y = Abundance))+
  geom_bar(stat="identity", position = "stack") +
  facet_grid(~year, scale="free")+
  labs(title="", subtitle="")  
```
## 5.2.2 multivariate analysis
```{r ordination}
fanga_ecm_ordi_mdsjacc <- ordinate(as_binary_otu_table(fanga_ecm), method = "MDS", dist="jaccard")
fanga_ecm_ordi_mdsjacc <- plot_ordination(fanga_ecm, fanga_ecm_ordi_mdsjacc, type="samples", color="treatment",label="sample_id")+
  geom_point(size=2)+
  facet_wrap(~year)+
  # color and elipse
  scale_color_manual(values = c("clearcut" = "red",
                              "forest" = "darkgreen",
                              "thinned_burn" = "grey4",
                              "thinned_no_burn" = "orange"))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="position from CC"))+
  ggtitle(label = "MDS of samples from fangamon by year", subtitle="presence-absence (jaccard distance)")
fanga_ecm_ordi_mdsjacc

fanga_ecm_ordi_mdsbray <- ordinate(fanga_ecm, method = "MDS", dist="bray")
fanga_ecm_ordi_mdsbray <- plot_ordination(fanga_ecm, fanga_ecm_ordi_mdsbray, type="samples", color="treatment",label="sample_id")+
  geom_point(size=2)+
  facet_wrap(~year)+
  # color and elipse
  scale_color_manual(values = c("clearcut" = "red",
                              "forest" = "darkgreen",
                              "thinned_burn" = "grey4",
                              "thinned_no_burn" = "orange"))+
  stat_ellipse(type="t", level=0.70)+
  # legend
  guides(color=guide_legend(title="position from CC"))+
  ggtitle(label = "MDS of samples from fangamon by year",subtitle = "abundance (bray-curtis distance)")
```


# *6 PRES_RESULTS 200323*
```{r SAMPLES SUPPRESSED}
miss_sam_clust_ecm$sample_id # total excluded
print(miss_sam_clust_ctrl) 
# every control and NA suppressed (ie 0 occurence) 
## 8 PCR neg => good control,
### what about ITS standard ? and Q ? 
### ITS probably allows us to see if amplification has been done right (for further comparisons)
print(miss_sam_clust_hala) 
print(miss_sam_clust_hala5)
print(miss_sam_clust_fanga)
### samples with 0 values... real absence or methods pb? to consider into study? 
```
```{r overview sites}
circle_pq(ecm_phy, "site", taxa = "Phylum", add_nb_seq = F,
          grid_col = "darkgrey",
          row_col = c("darkred","#1F78B4","darkgreen"))

read_distrib # density plot of nb of reads per site 
hist_cluster_size # distribution of the cluster size among ECM OTUs
ecm_phy_rrcurve # rarefaction curve for samples per site
sac_f_raw # cluster size accumulation curve from all fungi clusters
```
```{r hala edge}
# design
table(hala_sam$plot,hala_sam$treatment)

# number of genus
as.factor(hala_ecm@tax_table[,5]) %>% droplevels() %>% levels() 

hala_rrcurve_groups1
hala_rrcurve_groups2

circle_pq(physeq = hala_ecm, 
                              fact = "specification", 
                              taxa = "Genus", 
                              min_prop_mod=0.001,
                              min_prop_tax=0.01,
                              row_col = c("CC_edge" = "orange",
                              "CC_middle" = "red",
                              "FOR_middle" = "darkgreen"),
                              grid_col= 1,
                              add_nb_seq = T) # distribution circle sample - taxonomy 

# hill numbers
hala_hill0  # OTU richness
hala_hill1 # ie shannon index
hala_hill2 # ie simpson index
hala_hill_treatment[[4]] # tukeyHSD test

hala_ecm_ggplot1 # distribution genus (0;1)
hala_ecm_ggplot2 # distribution genus (nb of reads)
hala_ecm_ggplot3 # OTU count (0,1)
hala_ecm_ggplot4 # OTU abundance (nb reads)

hala_ecm_ordi_mdsjacc # MDS jaccard (0;1)
hala_ecm_ordi_mdsbray # MDS bray-curtis (nb of reads)
```
```{r fangamon}
# design
table(fanga_sam$plot,fanga_sam$treatment)

# number of genus
as.factor(fanga_ecm@tax_table[,5]) %>% droplevels() %>% levels() 

fanga_rrcurve0 # rarefaction curve for groups in 2012
fanga_rrcurve1 # rarefaction curve for groups in 2012

circle_fanga_ecm <- circle_pq(physeq = fanga_ecm, 
                              fact = "treatment", 
                              taxa = "Genus", 
                              min_prop_mod=0.001,
                              min_prop_tax=0.02,
                              row_col = c("clearcut" = "red",
                                "forest" = "darkgreen",
                                "thinned_burn" = "grey4",
                                "thinned_no_burn" = "orange"),
                              grid_col= 1,
                              add_nb_seq = T)

# hill numbers
fanga_hill0  # OTU richness # cc 2018 huge OTU richness for a single sample (can be seen also in distribution plots)
fanga_hill1 # ie shannon index
fanga_hill2 # ie simpson index
fanga_hill_treatment[[4]] # tukeyHSD test

fanga_ecm_ggplot1 # distribution genus (0;1)
fanga_ecm_ggplot2 # distribution genus (nb of reads)
fanga_ecm_ggplot3
fanga_ecm_ggplot4
# how to explained such results ? 
## a few OTUs explain everything
print(fanga_ecm_only_cc@otu_table)
which(fanga_ecm_only_cc@otu_table>200,arr.ind = T) %>% rownames() 
fanga_ecm_only_cc@tax_table[c("c10", "c21"),6] # 2 super abundant species 

### seedlings in just one core sampling -> pooled with other => rise the abundance of whole sample 
### affinity in the amplification ? maybe for c21 (only present in one sample) but not for c10 where it is more distributed and between 19 and 965 reads
### not that meaningful -> shoudl check about qpcr to see real biological activity 

fanga_ecm_ordi_mdsjacc # MDS jaccard (0;1)
fanga_ecm_ordi_mdsbray # MDS bray-curtis (nb of reads)
```
```{r hala5}
# design
table(hala5_sam$plot,hala5_sam$treatment)

# number of genus
as.factor(hala5_ecm@tax_table[,5]) %>% droplevels() %>% levels() 

hala5_rrcurve0 # rarefaction curve for samples
circle_pq(physeq = hala5_ecm, 
                              fact = "treatment", 
                              taxa = "Genus", 
                              min_prop_mod=0.001,
                              min_prop_tax=0.01,
                              row_col = c("clearcut" = "red",
                                          "forest" ="darkgreen",
                                          "thinned" = "orange"),
                              grid_col= 1,
                              add_nb_seq = T) # distribution circle sample - taxonomy 

# hill numbers 
hala5_hill0  # OTU richness
hala5_hill1 # ie shannon index
hala5_hill2 # ie simpson index
hala5_hill_treatment[[4]] # tukeyHSD test

hala5_ecm_ggplot1 # distribution genus (0;1)
hala5_ecm_ggplot2 # distribution genus (nb of reads)
hala5_ecm_ggplot3
hala5_ecm_ggplot4

hala5_ecm_ordi_mdsjacc # MDS jaccard (0;1)
hala5_ecm_ordi_mdsbray # MDS bray-curtis (nb of reads)
hala5_ecm_ordi_phyl1
```
